File: backend/src/main/java/com/backend/domain/community/dto/response/CommentResponseDTO.java
package com.backend.domain.community.dto.response;

import com.backend.domain.community.entity.Comment;

import java.time.LocalDateTime;

public record CommentResponseDTO(
        Long commentId,
        Long memberId,
        String name,
        String comment,
        LocalDateTime createDate,
        boolean deleted
) {
    public CommentResponseDTO(Comment comment, String userName) {
        this(
                comment.getId(),
                comment.getMemberId(),
                userName,
                comment.getComment(),
                comment.getCreateDate(),
                comment.isDeleted()
        );
    }
}

File: backend/src/main/java/com/backend/domain/community/dto/response/CommentResponseDTO.kt
package com.backend.domain.community.dto.response

import com.backend.domain.community.entity.Comment
import com.backend.domain.user.entity.User
import java.time.LocalDateTime

data class CommentResponseDTO(
    val commentId: Long?,
    val memberId: Long?,
    val name: String,
    val comment: String,
    val createDate: LocalDateTime?,
    val deleted: Boolean,
    val userImage: String?
) {
    constructor(comment: Comment, user: User) : this(
        comment.id,
        comment.memberId,
        user.name,
        comment.comment,
        comment.createDate,
        comment.deleted,
        user.imageUrl
    )
}